include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +512,74 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class ForwardIterator, class OutputIterator>
+auto sorted_mode(ForwardIterator first, ForwardIterator last, OutputIterator output) -> decltype(output)
+{
+    using Z = typename std::iterator_traits<ForwardIterator>::value_type;
+    static_assert(std::is_integral<Z>::value, "Floating point values have not yet been implemented.");
+    using Size = typename std::iterator_traits<ForwardIterator>::difference_type;
+
+    std::vector<Z> modes {};
+    modes.reserve(16);
+    Size max_counter {0};
+    auto it {first};
+
+    if(first == last)
+    {
+        return output;
+    }
+    
+    while(it != last)
+    {
+        Size current_count {0};
+        auto end_it {it};
+        while(end_it != last && *end_it == *it)
+        {
+            ++current_count;
+            ++end_it;
+        }
```

Comment:
You don't have to increment `current_count` on every iteration of this loop. You can just loop until the condition is met and then `current_count += std::distance(it, end_it)`.

And then, you'll see that this loop is just `find_if` with a predicate for 'is not equal to `*it`'.

Reply:
Ahh, actually, if the input sequence is not random access, then `std::distance` will be linear. Hmm. Maybe best to leave it as is after all. :)